tests/kv679_tombstone.erl

```diff
@@ -0,0 +1,227 @@
+
+write_key(_, [], _Opts) ->
+    ok;
+write_key(Client, [Val | Rest], Opts) ->
+    ok = write_key(Client, Val, Opts),
+    ok = write_key(Client, Rest, Opts);
+write_key({_, Client}, Val, Opts) when is_binary(Val) ->
+    Object = case riakc_pb_socket:get(Client, ?BUCKET, ?KEY, []) of
+                 {ok, O1} ->
+                     lager:info("writing existing!"),
+                     riakc_obj:update_value(O1, Val);
+                 _ ->
+                     lager:info("writing new!"),
+                     riakc_obj:new(?BUCKET, ?KEY, Val)
+             end,
+    riakc_pb_socket:put(Client, Object, Opts).
+
+read_key({_, Client}) ->
+    riakc_pb_socket:get(Client, ?BUCKET, ?KEY, []).
+
+delete_key({_, Client}) ->
+    riakc_pb_socket:delete(Client, ?BUCKET, ?KEY),
+
+    rt:wait_until(fun() ->
+                          case  riakc_pb_socket:get(Client, ?BUCKET, ?KEY, [deletedvclock]) of
+                              {error, notfound, VC} ->
+                                  lager:info("TSVC ~p~n", [VC]),
+                                  true;
+                              Res ->
+                                  lager:info("no ts yet: ~p~n", [Res]),
+                                  false
+                          end
+                  end).
+
+read_it_and_reap({_, Client}) ->
+    rt:wait_until(fun() ->
+                          case  riakc_pb_socket:get(Client, ?BUCKET, ?KEY, [deletedvclock]) of
+                              {error, notfound} ->
+                                  true;
+                              Res ->
+                                  lager:info("not reaped ts yet: ~p~n", [Res]),
+                                  false
+                          end
+                  end).
+
+%%% Node ops
+
+start_node(Node, Preflist) ->
+    rt:start_and_wait(Node),
+    wait_for_new_pl(Preflist, Node).
+
+get_preflist(Node) ->
+    Chash = rpc:call(Node, riak_core_util, chash_key, [{?BUCKET, ?KEY}]),
+    UpNodes = rpc:call(Node, riak_core_node_watcher, nodes, [riak_kv]),
+    PL = rpc:call(Node, riak_core_apl, get_apl_ann, [Chash, 3, UpNodes]),
+    PL.
+
+kill_primary(Preflist) ->
+    kill_and_wait(Preflist, primary).
+
+kill_fallback(Preflist) ->
+    kill_and_wait(Preflist, fallback).
+
+kill_and_wait(Preflist, Type) ->
+        case lists:keytake(Type, 2, Preflist) of
+        false ->
+            erlang:error(no_nodes_of_type, [Type, Preflist]);
+        {value, {{Idx, Node}, Type}, PL2} ->
+            kill_node(Node),
+            lager:info("killed ~p~n", [Node]),
+            [{{_, N2}, _}|_] = PL2,
+            {wait_for_new_pl(Preflist, N2), Node, Idx}
+    end.
+
+kill_node(Node) ->
+    rt:stop_and_wait(Node).
+
+wait_for_new_pl(PL, Node) ->
+    rt:wait_until(fun() ->
+                          NewPL = get_preflist(Node),
+                          lager:info("new ~p~n old ~p~nNode ~p~n", [NewPL, PL, Node]),
+                          NewPL /= PL%% andalso contains_fallback(NewPL)
```

Comment:
I agree that you don't need the `andalso contains_fallback(NewPL)`. In this case the comment should be removed and so should `contains_fallback/1`.